engine/runtime/src/main/java/org/enso/interpreter/runtime/data/ArrayOverBuffer.java

```diff
@@ -38,4 +50,49 @@ long getArraySize() {
   public static ArrayOverBuffer wrapBuffer(ByteBuffer buffer) {
     return new ArrayOverBuffer(buffer);
   }
+
+  @ExportMessage
+  String toDisplayString(boolean allowSideEffects) {
+    final InteropLibrary iop = InteropLibrary.getUncached();
+    StringBuilder sb = new StringBuilder();
+    try {
+      sb.append('[');
+      String sep = "";
+      long len = getArraySize();
+      for (long i = 0; i < len; i++) {
+        sb.append(sep);
+
+        Object at = readArrayElement(i);
+        Object str = showObject(iop, allowSideEffects, at);
+        if (iop.isString(str)) {
+          sb.append(iop.asString(str));
+        } else {
+          sb.append("_");
+        }
+        sep = ", ";
+      }
+      sb.append(']');
+    } catch (InvalidArrayIndexException | UnsupportedMessageException ex) {
+      StringWriter w = new StringWriter();
+      ex.printStackTrace(new PrintWriter(w));
+      sb.append("...\n").append(w.toString());
+    }
+    return sb.toString();
+  }
+
+  @CompilerDirectives.TruffleBoundary
+  private Object showObject(InteropLibrary iop, boolean allowSideEffects, Object child)
+      throws UnsupportedMessageException {
+    if (child == null) {
+      return "null";
+    } else if (child instanceof Boolean) {
+      return (boolean) child ? "True" : "False";
+    } else {
```

Comment:
I'm really worried that this logic for converting true/false to their Enso-textual representation is scattered throughout the project. It's in 3 places already and this instance is fourth.

I know that this logic is unlikely to change, so it's probably not a problem, but just so we can locate that - can we at least fold the `"True"` and `"False"` into some constants?

Still, ideally I think we should encapsulate this whole `showObject` logic (i.e. the part checking for `null`, for Boolean and then delegating to `iop`) into a separate function. It seems really unclean to repeat it over and over again.

Reply:
Other question is - in this particular scenario - is this code even needed?

I'd find it acceptable to just delegate to a common 'handle showObject' code and not think about particular cases as it makes the code simplest to reason about.

But here I think we have code visualizing a _byte_ array, which is written to handle booleans. I don't think we can have booleans in the byte buffer.

Still, I think it's better to just move this into a common function and call that one in all cases (handling booleans). But if we were to not do that, let's at least remove the special case for booleans which is in practice dead code here.

Reply:
True, we can simplify it further. Let me take a look.